packages/x-license/src/utils/licenseInfo.ts

```diff
@@ -29,5 +33,15 @@ export class LicenseInfo {
   public static setLicenseKey(key: string) {
     const licenseInfo = LicenseInfo.getLicenseInfo();
     licenseInfo.key = key;
+    LicenseInfo.isLicenseKeySet = !!key;
+    LicenseInfo.hasSetLicenseKeyRun = true;
+  }
+
+  public static getIsLicenseKeySet(): boolean {
+    return LicenseInfo.isLicenseKeySet;
+  }
+
+  public static getHasSetLicenseKeyRun(): boolean {
+    return LicenseInfo.hasSetLicenseKeyRun;
```

Comment:
We could do this with less code, no?

```suggestion
  }
```

and then 

```diff
 ponyfillGlobal.__MUI_LICENSE_INFO__ = ponyfillGlobal.__MUI_LICENSE_INFO__ || {
-  key: undefined,
+  key: 'default',
 };
```

Reply:
There is one very specific edge-case where this would not be enough:
`setLicenseKey` was called, but with `undefined` as parameter.
Not sure if we want to cover that case though ... WDYT?